Community/ideAlarm/automation/lib/python/community/idealarm/__init__.py

```diff
@@ -0,0 +1,504 @@
+# -*- coding: utf-8 -*-
+
+import weakref # Using this to prevent problems with garbage collection
+
+from core.jsr223 import scope
+from core.log import logging, LOG_PREFIX
+from core.utils import isActive, getItemValue, postUpdateCheckFirst, sendCommandCheckFirst, kw
+from community.idealarm import custom
+
+from org.eclipse.smarthome.model.persistence.extensions import PersistenceExtensions
+
+from org.joda.time import DateTime
+
+log = logging.getLogger(LOG_PREFIX + '.ideAlarm')
+
+ZONESTATUS = {'NORMAL': 0, 'ALERT': 1, 'ERROR': 2, 'TRIPPED': 3, 'ARMING': 4}
+ARMINGMODE = {'DISARMED': 0, 'ARMED_HOME': 1, 'ARMED_AWAY': 2}
+
+class IdeAlarmError(Exception):
+    '''
+    Base class for IdeAlarm errors
+    '''
+
+class IdeAlarmSensor(object):
+    '''
+    Alarm Sensor Object
+    '''
+    def __init__(self, parent, cfg):
+        '''
+        Initialise the IdeAlarmSensor class
+
+        Expects:
+         - Parent object
+         - cfg (dictionary) The sensor's configuration dictionary
+        '''
+        self.name = cfg['name']
+        _label = itemRegistry.getItem(self.name).label
+        self.label = _label if _label is not None else 'Sensor has no label'
+        self.parent = weakref.ref(parent) # <= garbage-collector safe!
+        self.sensorClass = cfg['sensorClass']
+        self.nag = cfg['nag']
+        self.nagTimeoutMins = cfg['nagTimeoutMins']
+        self.armWarn = cfg['armWarn']
+        self.enabled = cfg['enabled']
+        self.log = logging.getLogger("{}.IdeAlarmSensor.{}".format(LOG_PREFIX, self.name))
+        #self.log.info("ideAlarm sensor {} initialized...".format(self.name))
```

Comment:
I suspected this would be an issue when replacing the + operators with format. I didn't get a chance to test it. We could put `from __future__ import unicode_literals` at the start of the script, but I think it is cleaner changing the strings to unicode.

> `self.log = logging.getLogger(u"{}.IdeAlarmZone.{}".format(LOG_PREFIX, self.name.decode('utf8')))` doesn't work as I intended though.

I haven't found a way to use unicode in the Logger. Is this something new that you're trying to get to work, or was this working before? It doesn't work using the + operator either.

Reply:
> I suspected this would be an issue when replacing the + operators with format. I didn't get a chance to test it. We could put `from __future__ import unicode_literals` at the start of the script, but I think it is cleaner changing the strings to unicode.
> 
yep so do I! Later I will convert all my personal scripts to use the "format style" too.

> > `self.log = logging.getLogger(u"{}.IdeAlarmZone.{}".format(LOG_PREFIX, self.name.decode('utf8')))` doesn't work as I intended though.
> 
> I haven't found a way to use unicode in the Logger. Is this something new that you're trying to get to work, or was this working before? It doesn't work using the + operator either.

I thought it was working before but now when I have started OH2.3 again, I see that it's the same issue, that is, we haven't broken anything and it's just cosmetic anyway.  ;)